### ISSUE TITLE
Documentation on keyfile migration

### ISSUE THREAD

**Issue by ky-ex** [keepassxreboot/keepassxc#7216]
I've been a keepassX user, considering to migrate to keepassXC, because I saw multiple recommendations.

I'm using a password plus a keyfile. The keyfile is in a file format that starts with
```<?xml version="1.0" encoding="UTF-8"?><KeyFile><Meta><Version>1.00</Version></Meta><Key><Data>```
followed a string that looks like it's base64 encoded.

I need to access my database files from both desktop computers and Android devices, I'm using keepass2android.

Problem:
=======
keepassXC gives me a warning, legacy key file format. I want to resolve this issue prior to migrating, but I couldn't find any documentation or discussions specifically about this topic.

Question 1:
========
I noticed that the key file produced by keepassXC is bigger than the result of my conversion. Does keepassXC require a minimum key size? Is the data in the binary key file simply random data, or does it follow a structure?

Question 2:
========
Is it possible to keep using the same key, and simply convert the file format? If yes, could you please explain how to do the conversion?

The simply aproach didn't work. I tried to take the base64 string, decode it to binary and store it in a file. But if I use the binary result of the conversion as the key file in keepassXC, I get an error, file cannot be opened because wrong key.

Question 3:
=========
If a conversion isn't possible, then I probably must remove the old key file, and create a new one. If I do that, will the new key file also be compatible with keepass2android?

Thank you

**Comment by ky-ex** [comment]
Oh sorry. I must have made a mistake in my initial attempt.

I tried again. As I said, I decoded the base64 string inside the ```<data>...</data>``` tag, ran a tool to convert base64 to binary data, and saved the binary data to a file.

Then I used keepassXC to open the old database, with the new binary file.

That worked, it can decrypt and open.

However, I still get a warning, complaining about a "legacy file format".

I conclude that the warning is incorrect.

Apparently it isn't (just?) the file format that keepassXC is complaining about, but it seems to consider the small key size as legacy.

Is this intended?

I don't want to do the upgrade-key-file dance on all the devices that I currently have the various keepass tools installed (for me and my partner).

If future versions of keepassXC will continue to support the binary file format with the smaller keyfile size, then I wouldn't be required to upgrade the key files at this time.
